### PR TITLE
fix(services): fix dockerfile validation

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1194

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There is no better time than now, better future does not exist

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] test-remi2 - fix(services): fix dockerfile validation - 2024-06-27T16:14:15Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/43923935-b13d-4888-862e-5af3baa28f15/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] test-remi2 - fix(services): fix dockerfile validation - 2024-06-27T16:14:15Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/43923935-b13d-4888-862e-5af3baa28f15/logs)

File: libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx [hunk 1]
```diff
@@ -81,7 +89,7 @@ export function StepDockerfileFeature() {
                 Back
               </Button>
               <div className="flex gap-3">
-                <Button type="submit" size="lg" disabled={isLoadingCheckDockerfile}>
+                <Button type="submit" size="lg" disabled={isLoadingCheckDockerfile || !isValid}>
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx:92]: Missing `loading` state to check if dockerfile is valid

File: libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx [hunk 2]
```diff
@@ -78,16 +77,22 @@ export function DockerfileSettings({ children, methods, onSubmit, directSubmit =
...
-          setFileSource(value as FileSource)
-        }}
+      <Controller
+        name="dockerfile_source"
+        control={methods.control}
+        defaultValue="GIT_REPOSITORY"
+        render={({ field }) => (
+          <InputSelect
+            label="File source"
+            value={field.value as string}
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx:87]: `as string` not necessary no ?

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/dockerfile-settings/dockerfile-settings.tsx]: Fixed with https://github.com/Qovery/console/pull/1459/commits/bd2624514654845e05ebb145d89ec34f2ebd9036

File: libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/feature/page-job-create-feature/step-dockerfile-feature/step-dockerfile-feature.tsx]: Fixed with https://github.com/Qovery/console/pull/1459/commits/bd2624514654845e05ebb145d89ec34f2ebd9036

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**